pkg/providers/aws/provider_postgres.go

```diff
@@ -657,7 +664,11 @@ func buildRDSUpdateStrategy(rdsConfig *rds.CreateDBInstanceInput, foundConfig *r
 		mi.DeletionProtection = rdsConfig.DeletionProtection
 		updateFound = true
 	}
-	if *rdsConfig.Port != *foundConfig.Endpoint.Port {
+	config := *foundConfig
+	endpoint := config.Endpoint
+	if endpoint == nil {
+		// nil pointer check as *foundConfig.Endpoint doesn't exist on first run
+	} else if *rdsConfig.Port != *foundConfig.Endpoint.Port {
```

Comment:
I think we can shorten this a bit.
```suggestion
	if *foundConfig.Endpoint != nil && *rdsConfig.Port != *foundConfig.Endpoint.Port {
```

Reply:
compiler doesn't like it type issue `Cannot convert 'nil' to type 'Endpoint'` , may be able to shorten a little.